yolact.py

```diff
@@ -462,40 +462,54 @@ def __init__(self):
             # This comes from the smallest layer selected
             # Also note that cfg.num_classes includes background
             self.class_existence_fc = nn.Linear(src_channels[-1], cfg.num_classes - 1)
-        
+
         if cfg.use_semantic_segmentation_loss:
             self.semantic_seg_conv = nn.Conv2d(src_channels[0], cfg.num_classes-1, kernel_size=1)
 
         # For use in evaluation
         self.detect = Detect(cfg.num_classes, bkg_label=0, top_k=cfg.nms_top_k,
             conf_thresh=cfg.nms_conf_thresh, nms_thresh=cfg.nms_thresh)
 
-    def save_weights(self, path):
+    def save_checkpoint(self, path, recipe=None, epoch=0):
         """ Saves the model's weights using compression because the file sizes were getting too big. """
-        torch.save(self.state_dict(), path)
-    
-    def load_weights(self, path):
-        """ Loads weights from a compressed save file. """
-        state_dict = torch.load(path)
+        checkpoint = {
+                         'epoch': epoch,
+                         'model_state_dict': self.state_dict(),
+                         'recipe': recipe,
+                     },
+        torch.save(checkpoint, path)
+
+    def load_checkpoint(self, path) -> Tuple[Optional[int], Optional[str]]:
+        """ Loads weights into the current Yolact object.
+
+        :return: A tuple containing the epoch and the recipe if
+            found in the checkpoint
+        """
+
+        checkpoint = torch.load(path)
+        state_dict = checkpoint.get('model_state_dict', checkpoint)
+        epoch = checkpoint.get('epoch')
+        recipe = checkpoint.get('recipe')
```

Comment:
if `epoch` and `recipe` are both present here, then the recipe should be applied at the given epoch. (this will allow for flows like loading weights for a QAT model)

Reply:
functionality added in train.py!